simple_model.py

```diff
@@ -1,74 +1,80 @@
 import matplotlib.pyplot as plt
 import pandas as pd
-
+from sklearn.metrics import r2_score, mean_squared_error
 """
 HERE U EXPLAIN WHAT THIS SCRIPT IS FOR. 
 """
 
 
-def create_simple_ml_model(X, y, w, reg_):
+def create_simple_ml_model(X_train, X_test, y_train, w_train, reg_):
     """
-    helper function for make_simple_predictions_for_stacking(X, y, w, regressor, name, dict)
-
-    :param X: X variables (floats)
-    :param y: endogenous variable (float)
-    :param w: assignment to treatment (0 or 1)
-    :param regressor: a machine learning model
-    :return: the estimated treatment effect, as the difference between predicted outcome of treated and predicted outcome of untreated.
+    helper function for make_simple_predictions_for_stacking(X_train, X_test, y_train, w_train, regressors)
+
+    :param X_train: X variables (floats) (training set)
+    :param X_test: X variables (floats) (testing set)
+    :param y_train: endogenous variable (float) (training set)
+    :param w_train: assignment to treatment (0 or 1) (training set)
+    :param reg_: a machine learning model instance
+    :return: the estimated treatment effect for the testing set,
+        as the difference between predicted outcome of treated and predicted outcome of untreated.
     """
 
-    df = pd.DataFrame(data=X)
-    df['assignment'] = w
+    df = pd.DataFrame(data=X_train)
+    df['assignment'] = w_train
 
-    y_df = pd.DataFrame(data=y)
+    y_df = pd.DataFrame(data=y_train)
     reg = reg_.fit(df, y_df)
 
-    X_neg = pd.DataFrame(data=X)
+    X_neg = pd.DataFrame(data=X_test)
     X_neg['assignment'] = 0
 
-    X_pos = pd.DataFrame(data=X)
+    X_pos = pd.DataFrame(data=X_test)
     X_pos['assignment'] = 1
-    ret = reg.predict(X_pos) / reg.predict(X_neg)
+    ret = reg.predict(X_pos) - reg.predict(X_neg)
     return ret
 
 
-def make_simple_predictions_for_stacking(X, y, w, regressors):
+def make_simple_predictions_for_stacking(X_train, X_test, y_train, w_train, regressors):
     """
-    :param X: X variables (floats)
-    :param y: endogenous variable (float)
-    :param w: assignment to treatment (0 or 1)
+    :param X_train: X variables (floats) (training set)
+    :param X_test: X variables (floats) (testing set)
+    :param y_train: endogenous variable (float) (training set)
+    :param w_train: assignment to treatment (0 or 1) (training set)
     :param regressors: dictionary of name, regressor
     :return: dict of regressors (name as string), predictions
     """
     predictions_dict = {}
 
     for name in regressors.keys():
-        preds = create_simple_ml_model(X, y, w, regressors[name])
+        preds = create_simple_ml_model(X_train, X_test, y_train, w_train, regressors[name])
         predictions_dict[name] = preds
 
     return predictions_dict
 
 
 # TODO: add average line to plot, add true treatment effect to plot.
-def multilayer_hist(dict):
+def multilayer_hist(dictionary, true_vals):
```

Comment:
eh, i would put this in a new function or at least change the name of this considering it doesnt make a multilayer hist anymore